addon/components/ember-thead/component.js

```diff
@@ -281,15 +295,59 @@ export default Component.extend({
     this.columnTree.set('enableReorder', this.get('enableReorder'));
   },
 
+  _updateColumnMetaCache() {
+    this._validateUniqueColumnKeys();
+    this.columnMetaCache.keyPath = this.get('columnKeyPath');
+  },
+
   _onColumnsChange() {
     if (this.get('columns.length') === 0) {
       return;
     }
+    this._validateUniqueColumnKeys();
     this._updateColumnTree();
 
     scheduleOnce('actions', this, this.fillupHandler);
   },
 
+  _validateUniqueColumnKeys() {
```

Comment:
Should this be written in an `assert` style so that it doesn't run in production?

Reply:
I'm cool with that.

Reply:
Done.